compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/StatementGenerator.kt

```diff
@@ -110,8 +113,9 @@ class StatementGenerator(
             )
         }
 
+        val sourceElement = property.nameIdentifier ?: property
```

Comment:
```suggestion
        val sourceElement =
            if (context.extensions.debugInfoOnlyOnVariablesInDestructuringDeclarations) {
                property.nameIdentifier ?: property
            } else {
                property
            }
```
WIthout this change, Native debug tests (`kt42208 with variable`, `kt42208 with passing lambda to another function`) are still failing